src/main/java/hudson/plugins/resultscache/CachedResult.java

```diff
@@ -0,0 +1,11 @@
+package hudson.plugins.resultscache;
+
+import hudson.model.Result;
+
+public class CachedResult {
+    public Result result;
+    public Number build_number;
```

Comment:
Why not `Integer` instead of `Number`?

Reply:
Mostly for consistency and null-check purposes.

Reply:
The issue with using "Number" is that it is an abstract class in java. You can manage the null-check using the Integer boxed type and, at the same time, you are using an instantiable class.

Reply:
Apart from that, I would rename the field from `build_number` to `buildNumber` in order to follow java naming conventions. Do not forget to rename the get/set methods, too.